maestro-client/src/main/java/maestro/Maestro.kt

```diff
@@ -177,10 +182,31 @@ class Maestro(private val driver: Driver) : AutoCloseable {
             val hierarchyAfterTap = viewHierarchy()
 
             if (hierarchyBeforeTap != hierarchyAfterTap) {
-                LOGGER.info("Something have changed in the UI. Proceed.")
+                LOGGER.info("Something have changed in the UI judging by view hierarchy. Proceed.")
                 return
             }
 
+            val screenshotAfterTap: BufferedImage? = ImageIO.read(takeScreenshot().inputStream())
+            if (screenshotBeforeTap != null &&
+                screenshotAfterTap != null &&
+                screenshotBeforeTap.width == screenshotAfterTap.width &&
+                screenshotBeforeTap.height == screenshotAfterTap.height
+            ) {
+                val imageDiff = ImageComparison(
+                    screenshotBeforeTap,
+                    screenshotAfterTap
+                ).compareImages().differencePercent
+
+                if (imageDiff > 0.005) {
```

Comment:
Is it like a difference of more than 5%? 

nit: If yes maybe put that in the constant field name.

Reply:
0.5% in fact